### PR TITLE
[CN DateTimeV2] Filter out false positives from company/product names "今日头条" and "明日之后"

### LINKED ISSUE #1823: [CN DateTimeV2] Incorrect extracting "今日" from "今日头条" and "明日" from "网易手游《明日之后》" (filter?)
hi, how to fix this issue?

### PR DESCRIPTION
**Pull request by FeifanLiu98**
This PR fix #1823.
A filter is added only for exactly those two cases.
Later we will design a way for consumers to define their own filters.

### CODE REVIEW

File: .NET/Microsoft.Recognizers.Text.DateTime/Chinese/Extractors/ChineseMergedExtractorConfiguration.cs [hunk 1]
```diff
@@ -104,6 +104,30 @@ private List<ExtractResult> CheckDenyList(List<ExtractResult> extractResults, st
                     {
                         continue;
                     }
+
+                    if (endIndex <= (text.Length - 2))
+                    {
+                        tmpChar = text.Substring(endIndex, 2);
+
+                        // for case "今日头条"
+                        if (extractResult.Text.Equals("今日") && endIndex < text.Length && tmpChar.Equals("头条"))
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/Chinese/Extractors/ChineseMergedExtractorConfiguration.cs]: Why not use the ambiguity filter that you have introduced?

**Comment by FeifanLiu98** [comment]
Because all entities in AmbiguityFilterDict should be filtered directly in any situation. Let us take "早" for example, every single "早" entity should be filtered. As for "今日", only when it is followed by "头条" it shouldn't be recognized. Assuming there is two "今日" in one sentence "今日头条在今日开放", it should only filter one "今日" followed by "头条" and leave another one. If "今日头条" were added to AmbiguityFilterDict, every "今日" would be filtered.

File: .NET/Microsoft.Recognizers.Text.DateTime/Chinese/Extractors/ChineseMergedExtractorConfiguration.cs (hunk 1 above)
Comment (tellarin) [reply .NET/Microsoft.Recognizers.Text.DateTime/Chinese/Extractors/ChineseMergedExtractorConfiguration.cs]: @FeifanLiu98 can you please reply here in the review thread? It makes it much easier to track the discussion.
In English you can see some usages of AmbiguityFiltersDict that do exactly what you mention. I.e., if "今日" is detected, filter it if "今日头" matches.

**Comment by FeifanLiu98** [comment]
I am confused about why the build in Java will stop with no Info while the log print
[INFO] BUILD SUCCESS

**Comment by tellarin** [comment]
@FeifanLiu98, I'm not sure what you mean.
The 3 last build logs say "[INFO] BUILD FAILURE" on the Java build failure.

File: .NET/Microsoft.Recognizers.Text.Number/English/Extractors/NumberExtractor.cs [hunk 2]
```diff
@@ -59,14 +59,8 @@ private NumberExtractor(NumberMode mode, NumberOptions options)
 
             var ambiguityBuilder = ImmutableDictionary.CreateBuilder<Regex, Regex>();
 
-            // Do not filter the ambiguous number cases like 'that one' in NumberWithUnit, otherwise they can't be resolved.
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.Number/English/Extractors/NumberExtractor.cs]: Why remove this comment?

Comment (tellarin) [reply .NET/Microsoft.Recognizers.Text.Number/English/Extractors/NumberExtractor.cs]: The comment was incorrect?

File: JavaScript/packages/recognizers-number/src/number/french/extractors.ts [hunk 3]
```diff
@@ -40,12 +40,7 @@ export class FrenchNumberExtractor extends BaseNumberExtractor {
         // Add filter
         let ambiguityFiltersDict = new Array<RegExpRegExp>();
 
-        if (mode != NumberMode.Unit){
-
-            for (let [ key, value ] of BaseNumbers.AmbiguityFiltersDict){
-                ambiguityFiltersDict.push({ regExpKey: RegExpUtility.getSafeRegExp(key, "gs"), regExpValue: RegExpUtility.getSafeRegExp(value, "gs")})
-            }
-            
+        if (mode != NumberMode.Unit){
```
Comment (tellarin) [inline JavaScript/packages/recognizers-number/src/number/french/extractors.ts]: Nitpick: Can you fix the code standard and add the whitespaces between ) and { ?

File: Specs/Number/English/NumberModel.json [hunk 4]
```diff
@@ -2551,6 +2551,7 @@
   },
   {
     "Input": "2000 over 3",
+    "NotSupported":  "Python",
```
Comment (tellarin) [inline Specs/Number/English/NumberModel.json]: Why? How was it supported earlier?

File: Specs/Number/French/NumberModel.json [hunk 5]
```diff
@@ -1661,6 +1661,7 @@
   },
   {
     "Input": "Je veux un crédit de €1000 sur 3 ans",
+    "NotSupported": "python",
```
Comment (tellarin) [inline Specs/Number/French/NumberModel.json]: Why?

File: Specs/Number/French/NumberModel.json [hunk 6]
```diff
@@ -1661,7 +1661,7 @@
   },
   {
     "Input": "Je veux un crédit de €1000 sur 3 ans",
-    "NotSupported": "python",
+    "NotSupported": "java, python",
```
Comment (tellarin) [inline Specs/Number/French/NumberModel.json]: Why?

Comment (FeifanLiu98) [reply Specs/Number/French/NumberModel.json]: In my local test, this sentence is failed in Java and python. I tried to figure out what caused this error, but I got nothing...

**Pull request merged**